subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4773,6 +4784,8 @@ int lwm2m_parse_peerinfo(char *url, struct sockaddr *addr, bool *use_dtls, bool
 	tmp = url[off + len];
 	url[off + len] = '\0';
 
+	memcpy(desthostname, url + off, len);
```

Comment:
Who's responsibility would be to allocate buffer for `desthostname`? Plus copying w/o validating the destination buffer size is error-prone.

Perhaps it'd be better to only store a pointer to the hostname and it's length? The data is already stored in the Secrutity/FW Update objects implementation, we just need to know where to look for it.